### PR TITLE
[MAJOR] Use async/await

### PR DESCRIPTION
**Pull request by SivanMehta**
In preparation for consuming `@wrhs/extract-config` and `workers-factory@1.7.4`. This is a ***major*** bump because of the removal of the `build-one` route

- [x] `async/await` conversion of (most of) `lib/construct/index.js`
- [x] Make sure the tests still pass so ensure functional parity.
- [x] Remove the `build-one` route

### CODE REVIEW

File: lib/construct/index.js [hunk 1]
```diff
@@ -395,27 +393,28 @@ class Constructor extends EventEmitter {
...
+    // First see if this package has already been built for a different env, we
+    // should only build a single version once
+    //
+    try {
+      await this.checkAndDownload(spec, paths);
+    } catch (err) {
+      if (err.install) {
+        await this.packer.repack(spec, content, paths, statusWriter);
+      } else {
+        return next(err);
```
Comment (jcrugzz) [inline lib/construct/index.js]: lets just return the paths and get rid of the callback. Throw if there's an error and catch it to check if its `err.install` etc. So we aren't mixing promises and callbacks

File: lib/construct/builder.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const rmrf = require('./rmrf').async;
+
+class Builder {
```
Comment (jcrugzz) [inline lib/construct/builder.js:5]: this seems like a duplicate of cleaner?

**Review by jcrugzz — CHANGES_REQUESTED** [review]
Couple nits but otherwise this looks good

File: lib/construct/builder.js (hunk 2 above)
Comment (SivanMehta) [reply lib/construct/builder.js:5]: yep it is. Forgot to delete it

File: lib/construct/index.js (hunk 1 above)
Comment (SivanMehta) [reply lib/construct/index.js]: donezo

File: lib/construct/builder.js (hunk 2 above)
Comment (SivanMehta) [reply lib/construct/builder.js:5]: nevermind it's back in a new form

**Review by jcrugzz — APPROVED** [review]
Overall this looks good @SivanMehta. Do you want to merge this PR and then do the follow up refactor to integrate `@wrhs/extract-config`?

**Pull request merged**